.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+on: [ push, pull_request, workflow_dispatch ]
+jobs:
+  test:
+    name: test ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [
+          windows-latest,
+          # macos-latest, # cppfront is currently broken on AppleClang
+          ubuntu-latest,
+        ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      # System set-up
+      - uses: actions/checkout@v2
+      - uses: ilammy/msvc-dev-cmd@v1
+      - uses: lukka/get-cmake@latest
+
+      - name: Install GCC 11
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-11 g++-11
+          echo "CC=gcc-11" >> $GITHUB_ENV
+          echo "CXX=g++-11" >> $GITHUB_ENV
+
+      # Main cppfront library
+      - name: Configure cppfront
+        run: cmake -G Ninja -S . -B build/cppfront -DCMAKE_BUILD_TYPE=Release
+      - name: Build cppfront
+        run: cmake --build build/cppfront --config Release
+      - name: Install cppfront locally
+        run: cmake --install build/cppfront --config Release --prefix _local
+
+      # Regression tests
+      - name: Configure regression tests
+        run: >
+          cmake -G Ninja -S regression-tests -B build/regression-tests -DCMAKE_BUILD_TYPE=Release 
+          "-DCMAKE_PREFIX_PATH=${{github.workspace}}/_local"
+      - name: Build regression tests
+        run: cmake --build build/regression-tests --config Release
+      - name: Run regression tests
+        run: ctest --output-on-failure -C Release -j 4
```

Comment:
Why 4 jobs? Both the Windows and Linux runners have only 2 cores: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
Two is better, then, good point.